tests/integration/kubernetes/confidential/k8s-confidential.bats

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+IMAGE_REPO="ghcr.io/confidential-containers/test-container"
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	( [ "${KATA_HYPERVISOR}" == "firecracker" ] || [ "${KATA_HYPERVISOR}" == "dragonball" ] || [ "${KATA_HYPERVISOR}" == "qemu-tdx" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ] ) && skip "Test not supported yet."
+
+	pod_name="confidential-unencrypted-test"
+	container_name="confidential-unencrypted-container"
+	get_pod_config_dir
+
+	k8s_generate_service_yaml "${TEST_DIR}/${pod_name}.yaml" "${IMAGE_REPO}:unencrypted"
+	k8s_yaml_set_annotation "${TEST_DIR}/${pod_name}.yaml" "io.katacontainers.config.pre_attestation.enabled" "false"
+}
+
+@test "Test unencrypted confidential container launch success." {
+  # Start the service/deployment/pod
+  esudo kubectl apply -f "${TEST_DIR}/sev-unencrypted.yaml"
+  
+  # Retrieve pod name, wait for it to come up, retrieve pod ip
+  pod_name=$(esudo kubectl get pod -o wide | grep ${pod_name} | awk '{print $1;}')
+
+  # Check pod creation
+  kubectl wait --for=condition=Ready --timeout=20 pod "$pod_name"
+
+  pod_ip=$(esudo kubectl get pod -o wide | grep ${pod_name} | awk '{print $6;}')
+
+  print_service_info
+
+  if [ "${KATA_HYPERVISOR}" == "qemu-snp" ]; then
+	# Look for SEV enabled in container dmesg output
+	sev_enabled=$(ssh -i ${SSH_KEY_FILE} \
+		-o "StrictHostKeyChecking no" \
+		-o "PasswordAuthentication=no" \
+		-t root@${pod_ip} \
+		'dmesg | grep SEV' || true)
+
+	if [ -z "$sev_enabled" ]; then
+		>&2 echo -e "${RED}KATA CC TEST - FAIL: SEV is NOT Enabled${NC}"
+		return 1
+	else
+		echo "DMESG REPORT: $sev_enabled"
+		echo -e "${GREEN}KATA CC TEST - PASS: SEV is Enabled${NC}"
+	fi
+  fi
```

Comment:
Ouch, very SEV specific, this need to be moved to its own function and there we make it TEE specific.

Reply:
So I was thinking in fututre, this will have a switch case like structure where based on the `KATA_HYPERVISOR` variable, we can check the dmesg. Do you think I should still create a separate file which contains SEV specific code?

Reply:
My suggestion above, in the same thread that was started by Tobin, would be able to make this generic.

Also, please, ${GREEN}, ${RED}, etc, is not defined and I don't think we should be using them here.

Reply:
Used `DMESG_STRING` variable to make it generic.